### DISCUSSION TITLE
Stop truncating error messages

### DISCUSSION THREAD

**Discussion by bencroker** [craftcms/commerce#2438] [Ideas]
**What are you trying to do?**

When a payment fails, the payment gateway generally returns a long, detailed error message that is stored in `Gateway Message`. Unfortunately, Guzzle truncates the error message, hiding crucial text and making it difficult to understand why payments are failing. 

**What's your proposed solution?**

Storing the full error message should make troubleshooting errors easier.

Released in version [7.2](https://github.com/guzzle/guzzle/blob/master/CHANGELOG.md#720---2020-10-10), https://github.com/guzzle/guzzle/pull/2795 added support for passing a body summarizer to the http errors middleware, which would allow the truncation to be removed (or at the very least increased).

For Guzzle older than 7.2, instead of storing the exception message:

```php
$error = $exception->getMessage();
```

the full response can be stored:

```php
$error = $exception->getResponse()->getBody()->getContents();
```

**Comment by lukeholder** [comment]
@bencroker Which gateway is storing such a long error message? The message should be human readable, so not super long, and the full response (which would also contain the error) should be stored in the transaction response.

**Comment by bencroker** [comment]
Some error messages contain validation errors which span multiple lines, for example:
https://developer.worldpay.com/docs/access-worldpay/reference/errors#nested-errors

```json
{
  "errorName": "bodyDoesNotMatchSchema",
  "message": "The json body provided does not match the expected schema",
  "validationErrors": [
    {
      "errorName": "fieldMustBeNumber",
      "message": "Field at path must be a number",
      "jsonPath": "$.amount"
    },
    {
      "errorName": "fieldIsMissing",
      "message": "Field at path must be present",
      "jsonPath": "$.description"
    },
    {
      "errorName": "fieldHasInvalidValue",
      "message": "Payment Instrument type must be card/wallet",
      "jsonPath": "$.paymentInstrument.type"
    }
  ]
}
```
